rpcs3/ds4_pad_handler.cpp

```diff
@@ -479,8 +479,20 @@ void ds4_pad_handler::CheckAddDevice(hid_device* hidDevice, hid_device_info* hid
 	// Let's try getting 0x81 feature report, which should will return mac address on wired, and should error on bluetooth
 	std::array<u8, 64> buf{};
 	buf[0] = 0x81;
-	if (hid_get_feature_report(hidDevice, buf.data(), DS4_FEATURE_REPORT_0x81_SIZE) > 0)
+	if (const auto length = hid_get_feature_report(hidDevice, buf.data(), DS4_FEATURE_REPORT_0x81_SIZE); length > 0)
 	{
+		if (length != DS4_FEATURE_REPORT_0x81_SIZE)
+		{
+			// Controller may not be genuine. These controllers do not have feature 0x81 implemented and calibration data is in bluetooth format even in USB mode!
+			ds4Dev->unAuthentic = true;
+			LOG_WARNING(HLE, "DS4 controller may not be genuine. Workaround enabled.");
+
+			// Read feature report 0x12 instead which is what the console uses.
+			buf[0] = 0x12;
+			buf[1] = 0;
+			hid_get_feature_report(hidDevice, buf.data(), 16);
```

Comment:
Might as well add the size to the consts at the top of this file

Reply:
Interesting, the other constants in this file are lifted verbatim from the linux driver, but my stuff isn't present there so I had to figure out the sizes from HID descriptor dumps and I completely forgot about the consts. Will add it for completeness/documentation purposes.